### DISCUSSION TITLE
Room temperature precision

### DISCUSSION THREAD

**Discussion by muxa** [issalig/toshiba_air_cond#20] [Ideas]
When converting room temperature reading it's possible to preserve 0.5 precision provided by the data by using this logic:

```cpp
current_temperature = static_cast<float>(temp_data_byte) / 2.0 - 35.0;
```

Remote: `RBC-AMS41E`

**Comment by issalig** [comment]
I think mine does the same, isn't it?
https://github.com/issalig/toshiba_air_cond/blob/8519bd482951cf5d1ea8c6cb05e4b0ac429eaba5/air/toshiba_serial.hpp#L349

**Reply by muxa** [reply]
Ah, you are right, I was looking at the [target temp calculation](https://github.com/issalig/toshiba_air_cond/blob/8519bd482951cf5d1ea8c6cb05e4b0ac429eaba5/air/toshiba_serial.hpp#L339).

I am implementing the logic as an ESPHome component.

**Reply by muxa** [reply]
I've published the code for the ESPHome component: https://github.com/muxa/esphome-tcc-link
